beacon_node/eth1/src/error.rs

```diff
@@ -0,0 +1,47 @@
+use ethabi::Error as AbiError;
+use ssz;
+use web3;
+
+#[derive(Debug)]
+pub enum Error {
+    /// Errors that occur on interaction with an Eth1 node.
+    Web3Error(web3::error::Error),
+    /// Web3 request timeout.
+    Timeout,
+    /// Error on interacting with a contract on Eth1.
+    ContractError(web3::contract::Error),
+    /// Decoding error.
+    DecodingError,
+    /// Invalid parameters
+    InvalidParam,
```

Comment:
I think this error needs more detail. Perhaps:

`InvalidParam(String)`

This way we can include some description as to which parameter is invalid and why.

Reply:
Yup. Some of the error types didn't make sense. Fixed that!